sklearn/utils/estimator_checks.py

```diff
@@ -847,6 +864,98 @@ def _generate_sparse_matrix(X_csr):
         yield sparse_format + "_64", X
 
 
+def check_array_api_input(
+    name, estimator_orig, *, array_namespace, device=None, dtype="float64"
+):
+    """Check that the array_api Array gives the same results as ndarrays."""
+    try:
+        xp = __import__(array_namespace)
+        xp = sys.modules[array_namespace]
+    except ImportError:
+        raise SkipTest(
+            f"{array_namespace} is not installed: not checking array_api input"
+        )
+    try:
+        import array_api_compat  # noqa
+    except ImportError:
+        raise SkipTest(
+            "array_api_compat is not installed: not checking array_api input"
+        )
+
+    if array_namespace == "torch" and device == "cuda" and not xp.has_cuda:
+        raise SkipTest("test requires cuda, which is not available")
+
+    X, y = make_classification(random_state=42)
```

Comment:
To be consistent with the other estimator checks, we need to use `_enforce_estimator_tags_X` and `_enforce_estimator_tags_y`.